packages/flutter/lib/src/widgets/text_editing_intents.dart

```diff
@@ -221,6 +221,18 @@ class ExtendSelectionVerticallyToAdjacentPageIntent extends DirectionalCaretMove
   }) : super(forward, collapseSelection);
 }
 
+/// Extends, or moves the current selection from the current
+/// [TextSelection.extent] position to the closest paragraph boundary.
+class ExtendSelectionToParagraphBoundaryIntent extends DirectionalCaretMovementIntent {
+  /// Creates an [ExtendSelectionToParagraphBoundaryIntent].
+  const ExtendSelectionToParagraphBoundaryIntent({
+    required bool forward,
+    required bool collapseSelection,
+    required bool collapseAtReversal,
+    required bool continuesAtWrap,
```

Comment:
Looks like all the uses here are with `continuesAtWrap = true`. Is this needed? In the end, we also have `ExtendSelectionToLineBreakIntent` to wrap at line breaks.